src/cli.rs

```diff
@@ -979,12 +978,26 @@ pub fn key_gen<'a, C: CLI<'a>>(mut path: PathBuf) -> Result<(), KeystoreError<C:
     Ok(())
 }
 
-async fn repl<'a, L: 'static + Ledger, C: CLI<'a, Ledger = L>>(
+async fn keystore_for_test<'a, L: 'static + Ledger, C: CLI<'a, Ledger = L>>(
+    path: PathBuf,
     args: C::Args,
-) -> Result<(), KeystoreError<L>> {
-    let (storage, _tmp_dir) = match args.storage_path() {
-        Some(storage) => (storage, None),
-        None if !args.use_tmp_storage() => {
+) -> Result<Keystore<'a, C>, KeystoreError<L>> {
+    let mut loader = TrivialKeystoreLoader { dir: path };
+    let universal_param = Box::leak(Box::new(L::srs()));
+    let backend = C::init_backend(universal_param, args)?;
+    Keystore::<C>::new(backend, &mut loader).await
+}
+
+async fn keystore<'a, L: 'static + Ledger, C: CLI<'a, Ledger = L>>(
+    args: C::Args,
+) -> Result<(Keystore<'a, C>, Reader, SharedIO), KeystoreError<L>> {
+    let (mut io, reader) = match args.io() {
+        Some(io) => (io.clone(), Reader::automated(io)),
+        None => (SharedIO::std(), Reader::interactive()),
+    };
+    let storage = match args.storage_path() {
+        Some(storage) if !args.use_tmp_storage() => storage,
```

Comment:
Are we going to change how we use `use_tmp_storage`? The comment for this arg says:
> If `true`, create a temporary directory for storage instead of using [CLIArgs::storage_path].

So I think the logic should be:
* If `use_tmp_storage` is `true`: use a temporary path.
* Otherwise:
  * If `let Some(storage) = storage_path`: use the provided path.
  * Otherwise: use the `HOME` env.

If we indeed want to keep the current implementation, then perhaps update the comment?

Reply:
I reverted this back to the way it was before, there was no reason to change it in the first place.